### PR TITLE
revert: feature to use workspace id to get details

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
@toddgiguere @SirSpidey @Aashiq-J lets meet tomorrow to finalise this. Some questions to consider...

- We are requiring `vpc_id` - should / can we ask for name instead (can there be VPCs with same name in same resource group?)
- The variable descrption for `ssh_public_key` mentions "If `public_key` is not provided, the named key will be looked up from data." - what does this mean? I don't think that is correct description? We should be probably mentioning to use `existing_ssh_key_name` to lookup existing key name?
- The image name needs to be updated to `ibm-ubuntu-22-04-3-minimal-amd64-1` to match other VSI patterns (FYI I created an [issue](https://github.ibm.com/GoldenEye/issues/issues/6691) to updated our automation to bump this going forward)
- If we really need to ask for `boot_volume_encryption_key` we should update the descrption to mention where a consumer could find it (in which output of SLZ VPC DA). But can / should we ask for name?
- `existing_kms_instance_guid` - This is only needed if we are creating the auth policy (which we do not by default since its created in base SLZ VPC), so it should have default value of `null` and descprtion should be updated
- Do the default values of the `subnet_names` match the default values they will be called by the base SLZ VPC? I'm guessing not, so perhaps we don't have any default here?

We need a full review of all variable descriptions here

**Comment by Aashiq-J** [comment]
by default the value of `add_kms_block_storage_s2s` is true and the vpc pattern creates a s2s between block_storage and kms. The screenshot is from the vpc patterns output.
Screenshot 2023-11-30 at 9 03 47 AM

So will remove the variable `existing_kms_instance_guid` and `skip_iam_authorization_policy`.

**Comment by Aashiq-J** [comment]
Hi @SirSpidey ,

I have updated the README with the details on how to get the values for variable. Please let me know, which docs should  move those details.
And please review the variables descriptions, especially the once marked with TODO

**Comment by SirSpidey** [comment]
Hi @Aashiq-J I'm reviewing the strings but then stopped to experiment with putting the information in the pattern readme in a separate file. First draft is as its own topic: [Adding a VSI to your VPC landing zone deployable architecture](https://test.cloud.ibm.com/docs-draft/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi)

**Comment by SirSpidey** [comment]
@Aashiq-J I created https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/647 to merge to this PR so you (and others) can see the changes.

File: patterns/vsi-extension/README.md [hunk 1]
```diff
@@ -1,27 +1,7 @@
...
-Follow these steps to get the schematics workspace ID.
-
-1. Click the Navigation menu icon, and then click Schematics > Workspaces.
-1. Select the Workspace that is associated with landing zone VPC.
-1. Click the Settings.
-1. In the Details section, you can find the Workspace ID.
-
-Pass the Workspace ID to the `prerequisite_workspace_id` variable and pass the name of the VPC to the `existing_vpc_name` to choosse the name of the VPC to which the user wants to deploy the VSI.
-Please provide the Workspace ID for the prerequisite workspace and the name of the existing VPC to the `prerequisite_workspace_id` and `existing_vpc_name` variables respectively, to identify the VPC where you want to deploy the VSI.
+To deploy this architecture, you need the VPC ID, subnet names, and boot volume encryption key from your existing VPC landing zone deployable architecture. For information about finding the information, see [Adding a VSI to your VPC landing zone deployable architecture](https://test.cloud.ibm.com/docs-draft/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi).
```
[the remark this answers is not in the document]
Comment (ocofaigh) [reply patterns/vsi-extension/README.md:7]: I think we need a "Before you begin" section that mentions the block storage to KMS auth policy must exist before provisioning this DA. And that it would of been created by the VPC landing zone DA if `add_kms_block_storage_s2s` was set to true which it is by default.

[the remark this answers is not in the document]
Comment (ocofaigh) [reply patterns/vsi-extension/README.md:7]: that link is for IBM Cloud stage. remove the `test` from the URL and it will be a prod link

File: patterns/vsi-extension/variables.tf [hunk 2]
```diff
@@ -16,20 +16,19 @@ variable "prefix" {
...
 variable "vpc_id" {
-  description = "The ID of the VPC where the VSI will be created."
+  description = "The ID of the VPC where you want to deploy the VSI. [Learn more](https://cloud.ibm.com/docs-draft/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi)."
   type        = string
-  default     = null
 }
 
 variable "existing_ssh_key_name" {
-  description = "The ID of the VPC where the VSI will be created."
+  description = "The name of a public SSH key in the region where you want to deploy the VSI. [Learn more](https://cloud.ibm.com/docs/vpc?topic=vpc-ssh-keys). To create an SSH key, use the 'ssh_public_key' input instead."
```
Comment (ocofaigh) [inline patterns/vsi-extension/variables.tf:24]: I don't think we should say "To create an SSH key" - We are not creating one, we are adding an SSH key value to the VPC region. @SirSpidey feel free to suggest rewording here

File: patterns/vsi-extension/variables.tf [hunk 3]
```diff
@@ -39,7 +38,7 @@ variable "ssh_public_key" {
 }
 
 variable "resource_tags" {
-  description = "A list of tags to add to the VSI, block storage, security group, floating IP, and load balancer created by the module."
+  description = "A list of resource tags to apply to resources created by this module."
```
Comment (ocofaigh) [inline patterns/vsi-extension/variables.tf:41]: This is not a module, it is a deployable architecture or solution

File: patterns/vsi-extension/variables.tf [hunk 4]
```diff
@@ -51,63 +50,53 @@ variable "access_tags" {
...
 variable "vsi_per_subnet" {
-  description = "The number of VSI instances for each subnet."
+  description = "The number of virtual servers to create on each VSI subnet."
   type        = number
   default     = 1
 }
 
 variable "subnet_names" {
-  description = "The subnets to deploy the VSI instances to."
+  description = "The subnets where you want to deploy the VSI. If not specified, the VSI is deployed to all the subnets in the VPC. [Learn more](https://cloud.ibm.com/docs-draft/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi)."
```
Comment (ocofaigh) [inline patterns/vsi-extension/variables.tf:82]: Perhaps reword first sentence to "A list of subnet names where you want to deploy a VSI."

File: patterns/vsi-extension/variables.tf [hunk 5]
```diff
@@ -51,63 +50,53 @@ variable "access_tags" {
...
 variable "user_data" {
-  description = "User data to initialize VSI deployment."
+  description = "The user data to transfer to the instance. [Learn more](https://cloud.ibm.com/docs/vpc?topic=vpc-user-data)."
   type        = string
   default     = null
 }
 
 variable "boot_volume_encryption_key" {
-  description = "The CRN of the boot volume encryption key."
+  description = "The CRN of the boot volume encryption key. [Learn more](https://cloud.ibm.com/docs-draft/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi)."
```
Comment (toddgiguere) [inline patterns/vsi-extension/variables.tf:71]: I've noticed some of these docs links are "/docs-draft/" path, is that correct?

File: patterns/vsi-extension/README.md [hunk 6]
```diff
@@ -1,27 +1,16 @@
...
-## Using `vpc_id`
+This extension DA can be provisioned ontop of the VPC landing zone DA, as well as the ROKS landing zone DA since both deploy an SLZ VPC.
 
-The VPC ID of the landing zone VPC can be assigned to the variable vpc_id in order to create a VSI within that specific VPC.
+This architecture creates a VSI in some or all of the subnets of your existing landing zone VPC deployable architecture.
 
-## Using `prerequisite_workspace_id` and `existing_vpc_name`
+![Architecture diagram for adding a VSI to your VPC landing zone deployable architecture](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/reference-architectures/vsi-extension.drawio.svg)
 
-The user can specify the workspace ID associated with the deployment of the landing zone VPC when creating a new VSI.
```
Comment (toddgiguere) [inline patterns/vsi-extension/README.md:17]: @SirSpidey since we are removing the input for workspace, users who have deployed the VPC pattern via EPX/Project/Schematic/Catalog might need to know how to lookup some of the existing values that need to be supplied. We have some instruction links in the descriptions of the individual input variables, should we have something at this README level as well?

Comment (SirSpidey) [reply patterns/vsi-extension/README.md:17]: @toddgiguere we probably want to add the following kind of information to the `architecture overview` tab in the DA catalog ([like this](https://cloud.ibm.com/catalog/architecture/deploy-arch-ibm-pvs-inf-2dd486c7-b317-4aaa-907b-42671485ad96-global?catalog_query=aHR0cHM6Ly9jbG91ZC5pYm0uY29tL2NhdGFsb2cjcmVmZXJlbmNlX2FyY2hpdGVjdHVyZQ%3D%3D&kind=terraform&format=terraform&version=8e349dfd-a44d-413a-9aba-e7a228fd583e-global) for powervs with vpc extension). 

> To deploy this architecture, you need the VPC ID, subnet names, and boot volume encryption key from your existing VPC landing zone deployable architecture. For information about finding the information, see [Adding a VSI to your VPC landing zone deployable architecture](https://cloud.ibm.com/docs/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi).

This content is in the extension readme in this pr

Comment (SirSpidey) [reply patterns/vsi-extension/README.md:17]: @ocofaigh this comment above is more likely a comment for you, or whoever is onboarding the extension. I don't think the architecture overview tab is populated from the catalog json.

File: patterns/vsi-extension/variables.tf (hunk 5 above)
Comment (ocofaigh) [reply patterns/vsi-extension/variables.tf]: fixed

**Review by SirSpidey — APPROVED** [review]
👍 
None of the "learn more" links to cloud docs will work until you prepend `test.` to `cloud.ibm.com`

**Review by ocofaigh — APPROVED** [review]
